proxy/proxy/client.go

```diff
@@ -327,6 +328,12 @@ func needsRefresh(e cacheEntry, refreshCfgBuffer time.Duration) bool {
 }
 
 func (c *Client) cachedCfg(ctx context.Context, instance string) (string, *tls.Config, string, error) {
+	// Validate instance string in order to exclude potentially unwanted directories picked up by fuse.
+	if valid, err := util.Validate(instance); !valid {
```

Comment:
I'm not sure this is the best place to put the validation, but I need to read the fuse code some more to understand what the alternatives are.

Reply:
Looking some more, I see what's happening with fuse. Right now, the fuse code assumes all file system requests are valid Cloud SQL instances. That happens right here:

https://github.com/GoogleCloudPlatform/cloudsql-proxy/blob/02b93080ece77f30fe26212998c0f5f1c93e1ea1/proxy/fuse/fuse.go#L246-L248

I've filed a bug https://github.com/GoogleCloudPlatform/cloudsql-proxy/issues/873.

I think if you moved the call to `Validate` to just above line 246 returning an error if the name was invalid, we'd be able to close the bug, and we would have a nice solution overall. Want to submit that work as a separate PR first?